### ISSUE TITLE
[Reproduction] Request for the exact Qwen2.5-3B SEED w/o-SFT recipe (86.0 ALFWorld ablation)

### ISSUE THREAD

**Issue by zhangdw156** [jinyangwu/SEED#6]
Thank you for releasing SEED. I am trying to reproduce the paper's **w/o SFT** ablation, not the main SFT-initialized result, and would appreciate the exact missing recipe.

## Provenance and scope

I forked official [`jinyangwu/SEED`](https://github.com/jinyangwu/SEED) commit [`2cf2fad`](https://github.com/jinyangwu/SEED/commit/2cf2fadca3c5aba28da68e8e1405182ba8d90e6c) into [`zhangdw156/SEED`](https://github.com/zhangdw156/SEED/tree/exp/iclr). The branch history was snapshot-rewritten, but the root snapshot tree is identical to the official commit and the preserved content history descends from it. At experiment HEAD [`d52a24b`](https://github.com/zhangdw156/SEED/commit/d52a24b64ce62636b6169b89c0568eead5ce2236), the SEED advantage code, PPO/OPD loss, actor objective, analyzer, skill augmentation, and benchmark prompts are byte-identical to the official baseline. The fairness snapshot is [`3bfabe7`](https://github.com/zhangdw156/SEED/commit/3bfabe724df534155ce70626ecd95798e75a7c72).

The local w/o-SFT interpretation is: initialize Stage 2 from the original `Qwen2.5-{1.5B,3B}-Instruct` checkpoint while retaining the online analyzer, episode skill generation, skill-augmented re-scoring, OPD, and GRPO. This seems consistent with the paper description, but the public repository only contains SFT-initialized launchers and does not provide the exact Table 2 w/o-SFT command or checkpoint-selection rule.

## Why I added a fairness layer

The inherited environments could select different tasks across machines: ALFWorld depended on filesystem traversal and worker-local streams, while WebShop worker seeds could change the full content associated with a goal index. Independent validation workers could repeat tasks and omit others, and resumed jobs restarted the stream. I identified these behaviors through a local source audit and cross-machine checks, then tracked the fixes in the method fork history.

The fairness layer uses explicit canonical manifests: 3,553 ALFWorld training games, all 140 `val_seen`, all 134 `val_unseen`, 6,410 WebShop training goals, and 500 WebShop validation goals. It validates identities/counts, uses stable relative paths, makes WebShop goal construction worker-independent and fingerprint-checked, samples WebShop training goals without replacement before eight-rollout replication, and evaluates each fixed validation task exactly once in deterministic chunks. The 16 × 8 training group remains unchanged.

This is a controlled task protocol, not a claim that fairness caused the gap. A fairness-on/off factorial has not yet been run.

## Experiment protocol

- Raw Qwen2.5-Instruct initialization at 1.5B and 3B; online analysis and OPD remain enabled.
- 150 updates; task batch 16; rollout group 8; learning rate `1e-6`; clip 0.2; OPD beta 5 and coefficient 0.01; reference KL 0.01.
- ALFWorld: fixed 140-task `val_seen` panel; unweighted macro-average of six family success rates.
- WebShop: fixed 500-task panel; exact success.
- One stochastic validation sample per task at temperature 0.4, every five updates.
- One training seed per cell; four trainer GPUs.
- Local ALFWorld history/horizon are 2/50, while the official launcher uses 5/30. WebShop remains 2/15.
- GRPO uses the same fairness pools/evaluator but comes from a separate baseline implementation.

## Results

Paper Table 2 reports Qwen2.5-3B **SEED w/o SFT** ALFWorld family scores `97.6, 81.8, 92.6, 90.0, 84.2, 70.0`, macro `86.0`. My 3B step-150 macro is `70.0`, versus matched GRPO `65.9`; the local SEED AUC is `46.5`, versus GRPO `48.4`. The endpoint gain is therefore +4.1 rather than the paper's +11.0 over its GRPO result of 75.0.

| Scale | ALFWorld SEED w/o SFT / GRPO | WebShop SEED w/o SFT / GRPO |
|---|---:|---:|
| 1.5B | 71.5 / 61.9 | 63.8 / 62.8 |
| 3B | 70.0 / 65.9 | 71.4 / 63.6 |

There is no paper WebShop w/o-SFT result, so I do **not** treat the WebShop rows as a reproduction comparison. Likewise, I do not compare these raw-Instruct runs to the paper's main SFT-initialized SEED result (`91.8` ALFWorld and `78.9` WebShop success).

## Diagnostics

- All four SEED runs complete 150 updates with OPD enabled and no archived traceback, OOM, NaN, or resume.
- Analyzer success is 96.5% for ALFWorld 3B and 82.8% for ALFWorld 1.5B; failed analyses skip the teacher skill for that trajectory.
- Actor prompt clipping is zero in the archived runs.
- The 3B local deficit is concentrated in Heat (`56.25` versus paper `90.0`) and Pick2 (`41.67` versus `70.0`).
- The exact model/tokenizer revisions and task-file hashes were not recorded, and the fairness files were downloaded from an unpinned Hugging Face `main` revision; these remain provenance limitations.

## Questions

1. What exact initialization and command produced Table 2's Qwen2.5-3B w/o-SFT macro of 86.0? Was only the Stage-1 SFT checkpoint replaced by raw Instruct weights, as I assumed?
2. Could you provide the exact config overrides, code commit, model/tokenizer revision, ALFWorld task manifest, seed(s), and checkpoint-selection rule for that ablation?
3. Did the w/o-SFT paper run use 150 or the public launcher's 160 updates, and did it use ALFWorld history/horizon 5/30?
4. What analyzer parse/success rate did the paper run obtain, and should trajectories with failed analysis be skipped, retried, or trained with GRPO only?
5. Are the Table 2 values final-update, best-validation, or averaged results?
6. Is there an intended WebShop w/o-SFT configuration/result that can be used as a second reproduction check?

I can provide resolved configs, scalar CSVs, analyzer diagnostics, logs, and the complete source audit. A known-good w/o-SFT launcher would resolve the main ambiguity.

**Comment by zhangdw156** [comment]
<img width="2517" height="1714" alt="Image" src="https://github.com/user-attachments/assets/3d2c8460-8fd6-4673-8811-7908ca4db09f" />

**Comment by jinyangwu** [comment]
Thanks for the careful reproduction and source audit. Your interpretation is correct: “w/o Hindsight Skill SFT” initializes Stage 2 directly from the original Qwen2.5-3B-Instruct checkpoint, while retaining online trajectory analysis, episode-skill generation, skill-conditioned re-scoring, OPD, and GRPO. 

For the result reported in the paper, we used 150 policy updates, with history_length=5 and max_steps=30 for ALFWorld. The 160-update default in the released launcher is inconsistent with the paper configuration, and we will clarify/correct this. The Table 2 result corresponds to the evaluation at the final (150th) policy update, rather than the best validation checkpoint or an average over checkpoints.

When analyzer parsing fails, that trajectory receives no OPD teacher signal, but the standard GRPO objective is still applied.

Your history=2, max_steps=50 setting and modified task protocol therefore do not exactly match the paper run, so the 70.0 result is not directly comparable with Table 2’s 86.0. Nevertheless, the lower GRPO baseline and the Heat/Pick2 breakdown are useful diagnostics. 

The public release does not currently pin the exact scripts for this ablation, and we did not report a WebShop w/o-SFT result. We will check the archived experiment and provide a known-good launcher/config. Please feel free to share the resolved config and logs for comparison.
